sys/net/application_layer/nanocoap/sock.c

```diff
@@ -101,6 +86,37 @@ ssize_t nanocoap_get(sock_udp_ep_t *remote, const char *path, uint8_t *buf, size
     return res;
 }
 
+ssize_t nanocoap_get(sock_udp_ep_t *remote, const char *path, uint8_t *buf, size_t len)
+{
+    ssize_t res;
+    coap_pkt_t pkt;
+    uint8_t *pktpos = buf;
+
+    pkt.hdr = (coap_hdr_t*)buf;
+    pktpos += coap_build_hdr(pkt.hdr, COAP_REQ, NULL, 0, COAP_METHOD_GET, 1);
+    pktpos += coap_put_option_uri(pktpos, 0, path, COAP_OPT_URI_PATH);
+    pkt.payload_len = pktpos - buf;
+
+    res = nanocoap_request(&pkt, remote, len);
+    if (res < 0)
+    {
+        return res;
+    }
+    else {
+        res = coap_get_code(&pkt);
```

Comment:
IMHO better use `coap_get_code_class()` and then compare `res != COAP_CLASS_SUCCESS`

Reply:
oh, I see its copy paste from before and my suggestion would be an API change ... mhm

Reply:
Maybe in a follow up then?